app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt

```diff
@@ -50,8 +50,32 @@ class MessageMapper @Inject constructor(
             userList = userList
         )
 
-        val footer = if(message is Message.Regular) {
-            MessageFooter(message.id, message.reactions.totalReactions, message.reactions.selfUserReactions)
+        val footer = if (message is Message.Regular) {
+            // TODO find ugly and proper heart emoji and merge them to ugly one 😅
+            val totalHeartsCount = message.reactions.totalReactions
+                .filterKeys { isHeart(it) }.values
+                .scan(0) { acc, count -> acc + count }
+                .lastOrNull()
+            val hasSelfHeart = message.reactions.selfUserReactions.any { isHeart(it) }
+
+            MessageFooter(message.id,
+                message.reactions.totalReactions
+                    .filter { !isHeart(it.key) }
+                    .run {
+                        if (totalHeartsCount != null && totalHeartsCount != 0)
```

Comment:
With the suggestion above, totalHeartCount is no longer nullable.
```suggestion
                        if (totalHeartsCount != 0)
```